memote-report-app/src/app/report-data.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { ResultCard } from './resultcard.model';
+import { TestResult } from './test-result.model';
+
+@Injectable()
+export class ReportDataService {
+  metaData: any;
+  allTests: TestResult[] = [];
+  scoredCard: Object;
+  statisticsCards: ResultCard[] = [];
+
+  constructor(private http:HttpClient){
+  }
+
+  public loadResults(): void{
+    // this.http.get("data/testData.json").subscribe(data => {this.rawReportData = data});
+    this.http.get("data/testData.json").subscribe(data => {this.convertResults(data)});
+  }
+
+  public byID(string){
+    return this.allTests.find(x => x.id === string);
+  }
+
+  public getString(object){
+    return JSON.stringify(object);
+  }
+
+  private convertResults(data:Object): void{
+    // Store each test result as a TestResult object in a central list.
+    for (let test of Object.keys(data["tests"])){
```

Comment:
@ChristianLieven You should use `const test of ...` instead of `let test of ...` because you don't intend to reassign `test inside the for-block.